src/stratigraphy/util/layer_index_column.py

```diff
@@ -0,0 +1,162 @@
+        """Add a new layer index column entry to the layer index column.
+
+        Args:
+            entry (TextLine): The layer index column entry to be added.
+        """
+        self.entries.append(entry)
+
+    def can_be_appended(self, rect: fitz.Rect) -> bool:
+        """Checks if a new layer index column entry can be appended to the current layer index column.
+
+        The checks are:
+        - The width of the new rectangle is greater than the width of the current layer index column. Or;
+        - The middle of the new rectangle is within the horizontal boundaries of the current layer index column.
+        - The new rectangle intersects with the minimal horizontal boundaries of the current layer index column.
+
+
+        Args:
+            rect (fitz.Rect): Rect of the layer index column entry to be appended.
+
+        Returns:
+            bool: True if the new layer index column entry can be appended, False otherwise.
+        """
+        new_middle = (rect.x0 + rect.x1) / 2
+        if (self.rect().width < rect.width or self.rect().x0 < new_middle < self.rect().x1) and (
+            rect.x0 <= self.min_x1 and self.max_x0 <= rect.x1
+        ):
+            return True
+        return False
+
+    def strictly_contains(self, other):
+        return len(other.entries) < len(self.entries) and all(
+            other_entry in self.entries for other_entry in other.entries
+        )
+
+    def is_contained(self, rect: fitz.Rect) -> bool:
+        """Check if the layer index column is contained in another rectangle.
+
+        Args:
+            rect (fitz.Rect): The rectangle to check if it contains the layer index column.
+
+        Returns:
+            bool: True if the layer index column is contained in the rectangle, False otherwise.
+        """
+        return (
+            rect.x0 <= self.rect().x0
+            and self.rect().x1 <= rect.x1
+            and rect.y0 <= self.rect().y0
+            and self.rect().y1 <= rect.y1
+        )
+
+    def noise_count(self, words):
+        return 0
+
+
+def find_layer_index_column_entries(all_words: list[TextLine]) -> list:
+    r"""Find the layer index column entries.
+
+    Regex explanation:
+    - \b is a word boundary. This ensures that the match must start at the beginning of a word.
+    - [\da-z]+ matches one or more (+) alphanumeric characters (\d for digits and a-z for lowercase letters).
+    - \) matches a closing parenthesis. The backslash is necessary because parentheses are special characters
+      in regular expressions, so we need to escape it to match a literal parenthesis.
+    This regular expression will match strings like "1)", "2)", "a)", "b)", "1a4)", "6de)", etc.
+
+    Args:
+        all_words (ist[TextLine]): The words to search for layer index columns.
+
+    Returns:
+        list: The layer index column entries.
+    """
+    entries = []
+    for line in sorted(all_words, key=lambda line: line.rect.y0):
```

Comment:
Better name it "word" rather than line, even though it is of type TextLine (maybe this class should be renamed as well)?

Reply:
Are lines sometimes more than just a word? If not, then I am in favour of renaming the TextLine object.

And agree with naming it word over line.